index.js

```diff
@@ -1,9 +1,32 @@
 'use strict';
+const URL = require('url').Url;
 const urlRegex = require('url-regex');
 const normalizeUrl = require('normalize-url');
+const unescape = require('unescape');
 
 module.exports = (str, opts) => {
-	const urls = str.match(urlRegex());
-	const ret = urls ? urls.map(url => normalizeUrl(url.trim().replace(/\.+$/, ''), opts)) : [];
-	return new Set(ret);
+	opts = opts || {};
+	const ret = new Set();
+
+	function _extractQueryParams(url) {
+		const qs = new URL().parse(url, true).query;
+		for (const key in qs) {
+			if (Object.prototype.hasOwnProperty.call(qs, key) && qs[key].match(urlRegex())) {
+				_add(qs[key]);
+			}
+		}
+	}
+
+	function _add(url) {
```

Comment:
This can just be an arrow function instead. And don't prefix it with underscore.

Reply:
done